modules/affinity/src/main/java/org/apache/ignite/internal/affinity/AffinityManager.java

```diff
@@ -163,21 +113,43 @@ private void subscribeToAssignmentCalculation() {
     }
 
     /**
-     * Unsubscribes a listener form the affinity calculation.
+     * Calculates an assignment for a table which was specified by id.
+     *
+     * @param tblId Table identifier.
+     * @return A future which will complete when the assignment is calculated.
      */
-    private void unsubscribeFromAssignmentCalculation() {
-        if (affinityCalculateSubscriptionFut == null)
-            return;
-
-        try {
-            Long subscriptionId = affinityCalculateSubscriptionFut.get();
+    public CompletableFuture<Boolean> calculateAssignments(UUID tblId) {
+        return vaultManager.get(ByteArray.fromString(INTERNAL_PREFIX + tblId)).thenCompose(entry -> {
+            var tblConfig = configurationMgr.configurationRegistry().getConfiguration(TablesConfiguration.KEY).tables().get(new String(entry.value(), StandardCharsets.UTF_8));
+
+            var key = new Key(INTERNAL_PREFIX + tblId);
+
+            return metaStorageMgr.invoke(
+                Conditions.key(key).value().eq(null),
```

Comment:
Currently this will work, cause we don't support baseline changes, later we will need to generify
`Conditions.key(key).value().eq(null),`
to
`Conditions.key(key).value().eq(oldAssignmentsValue),`
where old assignments could be null, if it's table creation.